Issue: Adding Kubernetes container logs

Sahana-Bannigol-ML opened the pull request:
Root cause: Enhancement
Changes made:
i) Added flag named monitor_all_pods, if this flag is set then the associated project and app name from config are appended to the log record and pushed to snappyflow 
ii) Added code to extract projectname and appname from env
 iii) Added statements to free up memory
Testcases:
i)When monitor_all_pods flag is set, associated project name and app name is appended to the log record
 ii) When montir_all_pods flag is not set, log record remains as it is with the other fields.

<!-- Provide summary of changes -->

<!-- Issue number, if available. E.g. "Fixes #31", "Addresses #42, #77" -->

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change
<!-- Invoke Fluent Bit and Valgrind as: $ valgrind ./bin/fluent-bit <args> -->
- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

**Documentation**
<!-- Docs can be edited at https://github.com/fluent/fluent-bit-docs -->
- [ ] Documentation required for this feature

<!--  Doc PR (not required but highly recommended) -->

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

Commit:
Issue: Adding Kubernetes container logs
Root cause: Enhancement
Changes made:
i) Added flag named monitor_all_pods, if this flag is set then the associated project and app name from config are appended to the log record and pushed to snappyflow
ii) Added code to extract projectname and appname from env
iii) Added statements to free up memory
Testcases:
i)When monitor_all_pods flag is set, associated project name and app name is appended to the log record
ii) When montir_all_pods flag is not set, log record remains as it is with the other fields.

Sabari-Arunkumar-ML commented on plugins/filter_apm_kubernetes_labels/filter_kubernetes_labels.c:
```diff
@@ -39,6 +39,7 @@ static int configure(struct kubernetes_labels_ctx *ctx, struct flb_filter_instan
     if (file_path == NULL)
     {
         flb_error("Lookup key mapping_path not found in plugin configuration");
+        flb_free(file_path);
```
A NULL value should not be freed

Sabari-Arunkumar-ML commented on plugins/filter_apm_kubernetes_labels/filter_kubernetes_labels.c:
```diff
@@ -97,12 +98,34 @@ static int configure(struct kubernetes_labels_ctx *ctx, struct flb_filter_instan
+    else
+        ctx -> appname = DEFAULT_APPNAME;
+    
+    bool monitor_pods_label = getenv(MONITOR_ALL_PODS);
```
change monitor_pods_label variable name, this doesn't convey anything related to MONITOR_ALL_PODS flag

Sabari-Arunkumar-ML commented on plugins/filter_apm_kubernetes_labels/filter_kubernetes_labels.c:
```diff
@@ -97,12 +98,34 @@ static int configure(struct kubernetes_labels_ctx *ctx, struct flb_filter_instan
+    if(monitor_pods_label)
+        ctx -> monitor_pods_label = monitor_pods_label;
+    else
+        ctx -> monitor_pods_label = DEFAULT_MONITOR_PODS_LABEL;
```
change both  ctx -> monitor_pods_label, DEFAULT_MONITOR_PODS_LABEL variable accordingly

Sabari-Arunkumar-ML commented on plugins/filter_apm_kubernetes_labels/filter_kubernetes_labels.c:
```diff
@@ -212,10 +235,15 @@ static int cb_modifier_filter_apm_kubernetes_labels(const void *data, size_t byt
                 }
             }
         }
+        bool flag=true;
```
give proper variable name

Sabari-Arunkumar-ML commented on plugins/filter_apm_kubernetes_labels/filter_kubernetes_labels.c:
```diff
@@ -228,13 +256,14 @@ static int cb_modifier_filter_apm_kubernetes_labels(const void *data, size_t byt
             }
 
             if (t->start == -1 || t->end == -1 || (t->start == 0 && t->end == 0)){
+                flb_free(t);
```
dont free this variable, its just pointing to ctx->jsmn_tokens, which never should be freed

Sabari-Arunkumar-ML commented on plugins/filter_apm_kubernetes_labels/filter_kubernetes_labels.c:
```diff
@@ -291,6 +319,9 @@ static int cb_modifier_filter_apm_kubernetes_labels(const void *data, size_t byt
             }
 
         }
+        if (ctx-> monitor_pods_label){
```
this if check is done at line. 239, dont replicate it

Commit:
Issue: Adding Kubernetes container logs
Root cause: Enhancement
Changes made:
i) Changed naming of monitor_pods flag to monitor_pod_logs
ii) Renamed flag to readJson
iii) Removed two checks for montir_pods flag, combined it into one

Testcases:
i)When monitor_all_pods flag is set, associated project name and app name is appended to the log record
ii) When monitor_all_pods flag is not set, log record remains as it is with the other fields.
